velox/vector/FlatVector.cpp

```diff
@@ -198,8 +198,11 @@ void FlatVector<bool>::copyValuesAndNulls(
 template <>
 Buffer* FlatVector<StringView>::getBufferWithSpace(vector_size_t size) {
   // Check if the last buffer is uniquely referenced and has enough space.
+  //
+  // NOTE: begin() of 'stringBuffers_' (type of folly::F14VectorSet) points to
```

Comment:
Is this a guaranteed behavior?  Relying on the iteration order of hash table does not sound safe to me.

Reply:
I added a unit test to catch this potential behavior change on upgrade. Here we only need to track the first and last inserted element for flat vector usage, and the table has a linear array to track that. I added comment in this PR to disallow erase entry from the vector set.